src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from "react"
+import "./dropdown.scss"
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+
+
+export enum DropdownSelectType {
+    CHECKBOX = 'CHECKBOX',
+    LIST = 'LIST',
+}
+export interface DropdownProps {
+    options: { label: string, value: string }[];
+    defaultText: string;
+    type?: DropdownSelectType;
+    boxStyle?: {};
+    listStyle?: {};
+    disabled?: boolean;
+}
+
+const Dropdown = (props: DropdownProps) => {
+    // Set default to list if no style selected
+    let defaultDropdownType: DropdownSelectType = props.type as DropdownSelectType || DropdownSelectType.LIST;
+
+    const [isOpen, setIsOpen] = useState(true)
+    const [selectedOption, setSelectedOption] = useState(props.defaultText)
+
+    const updateSelection = (e: React.MouseEvent<HTMLLIElement>) => {
+        // console.log(e.target.value)
+        if (e.target instanceof HTMLElement) {
+            const targetVal = e.target.getAttribute('value')
+            console.log(targetVal)
+            setSelectedOption(e.target.textContent || '')
+            setIsOpen(false)
+        }
+    }
+
+    useEffect(() => {
+        setIsOpen(false)
+    }, [])
+    const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+        // console.log(e.currentTarget.value)
+        setSearchQuery('')
+        setIsOpen(!isOpen)
+    }
+
+    const clearSelection = () => {
+        setSelectedOption(props.defaultText)
+        setSearchQuery('')
+        setIsOpen(false)
+    }
+
+    const handleKeyDown = (event: React.KeyboardEvent) => {
+        // Allow users to click enter to select option
+        if (event.key === 'Enter') {
+            event.preventDefault();
+            const focusedElement = document.activeElement as HTMLElement;
```

Comment:
I actually wonder if React projects would prefer to completely manage focus themselves rather than rely on `document`? I genuinely don't know what's preferred.

Reply:
yep, move to reactful vs document interactions